### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by MasonSelf**

### CODE REVIEW

File: Character.cpp [hunk 1]
```diff
@@ -98,20 +99,47 @@ void Character::attackInternal(Character& other)
...
+        if (armor != *initialArmorLevel)
+        {
+            *initialArmorLevel *= 1.1;
+            armor = *initialArmorLevel;
+        }
+        if ( attackDamage != *initialAttackDamage)
+        {
+            *initialAttackDamage *= 1.1;
+            attackDamage = *initialAttackDamage;
+        }
```
Comment (matkatmusic) [inline Character.cpp:120]: All of this is basically doing the same thing 3 times.   
Use your knowledge of references to create a free function that you can call 3 times.
Also, 
```
a) your stats are restored to their initial value if they are lower than it.
```
emphasis on the `if they are lower than it`.   
your code is currently not checking for that.  
and you're not resetting your stat.  you're resetting the initial value to the current stat value.
- [ ] revise

File: DefensiveItem.cpp [hunk 2]
```diff
@@ -3,10 +3,13 @@
 #include "Paladin.h"
 #include "DragonSlayer.h"
 #include "Dragon.h"
-#include <assert>
+#include "Utility.h"
```
Comment (matkatmusic) [inline DefensiveItem.cpp:6]: Don't include what you don't use
- [ ] revise

File: Dragon.h [hunk 3]
```diff
@@ -2,9 +2,14 @@
 
 #include "Character.h"
 
-struct Dragon : Character
+struct Dragon : public Character
```
Comment (matkatmusic) [inline Dragon.h:5]: struct is public by default.  no need for this keyword
- [ ] revise

File: DragonSlayer.h [hunk 4]
```diff
@@ -1,8 +1,23 @@
 #pragma once
+#include "Character.h"
+#include "AttackItem.h"
+#include <memory>
 
-struct DragonSlayer : Character
+struct DragonSlayer : public Character
```
Comment (matkatmusic) [inline DragonSlayer.h:6]: struct is public by default.  no need for this keyword
- [ ] revise

File: Dwarf.h [hunk 5]
```diff
@@ -1,7 +1,15 @@
 #pragma once
+#include "Character.h"
 
-struct Dwarf : Character
+struct Dwarf : public Character
```
Comment (matkatmusic) [inline Dwarf.h:4]: struct is public by default.  no need for this keyword
- [ ] revise

File: HelpfulItem.h [hunk 6]
```diff
@@ -10,6 +10,6 @@
 struct HelpfulItem : Item
```
Comment (matkatmusic) [inline HelpfulItem.h:10]: Don't include what you don't use.  you're including a ton of header files in this file.
- [ ] revise

File: Paladin.h [hunk 7]
```diff
@@ -1,7 +1,13 @@
 #pragma once
+#include "Character.h"
 
-struct Paladin : Character
+struct Paladin : public Character
```
Comment (matkatmusic) [inline Paladin.h:4]: struct is public by default.  no need for this keyword
- [ ] revise

File: Utility.cpp [hunk 8]
```diff
@@ -43,66 +47,68 @@ std::string getCharacterStats(Character* ch)
...
+    else if( auto* chDS = dynamic_cast<DragonSlayer*>(character))
     {
-        assert(false);
+        //assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so their attack item should boost their attack damage by a factor of 10
         //this means you need to GET the attack damage, multiply it by the item's boost, and BOOST the attackDamage with that multiplied value.  
         //check Character.h for available member functions you can use.
+
+        chDS->boostAttackDamage(item->getBoost() * 10);
```
Comment (matkatmusic) [inline Utility.cpp:108]: this is not getting the attack damage and multiplying it by the boost. Take a look at the Character member functions available to you.
- [ ] revise

File: main.cpp [hunk 9]
```diff
@@ -70,12 +72,13 @@ it's very easy to mess this task up.
...
 {
+    
                       //name, hit points, armor. 
     Dwarf dwarf { "Maldin the Dwarf", 3, 1 };
     Paladin paladin { "Warrick the Paladin", 5, 2 };
     Dragon dragon { "Garry The Dragon", 200, 50 };
     DragonSlayer dragonSlayer { "Virgil the Dragon Slayer", 8, 5 };
-    
+    std::cout << "\ntest2" << std::endl;
     std::cout << "\nstart of battle" << std::endl;
```
Comment (matkatmusic) [inline main.cpp:82]: Verify that your output matches the expected output **exactly**. 
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 9 above)
Comment (MasonSelf) [reply main.cpp]: I put that in there trying to figure out if main() was even running. I addressed all these revisions, but am still getting "Killed exit status 137" as mentioned last week

File: Character.h [hunk 10]
```diff
@@ -35,6 +35,8 @@ struct Character
     
     const std::vector<std::unique_ptr<Item>>& getHelpfulItems() const { return helpfulItems; }
     const std::vector<std::unique_ptr<Item>>& getDefensiveItems() const { return defensiveItems; }
+
+    void victoryStatReset( int& currentStat, int& initialStat );
```
Comment (matkatmusic) [inline Character.h:39]: No one else is calling this function.  it does not need to be a member function.
make it a free function in the cpp file.
- [ ] revise

File: Dragon.cpp [hunk 11]
```diff
@@ -2,7 +2,7 @@
 #include "DragonSlayer.h"
 #include "Utility.h"
 
-Dragon::Dragon( std::string name_, int hitPoints, int armor ) : Character(hitPoints, armor, 80), name(name_) {}
+Dragon::Dragon( std::string name_, int hitPoints, int armor ) : Character(getHP(), armor, 80), name(name_) {}
```
Comment (matkatmusic) [inline Dragon.cpp:5]: `getHP()` is a member function of `Character`.  `Character` hasn't even been fully instantiated yet.  Take a look at the naming convention used by the ctor parameter that is initializing a Character member variable with the same name and follow that convention.
- [ ] revise

File: DragonSlayer.cpp [hunk 12]
```diff
@@ -3,7 +3,7 @@
 #include "Utility.h"
 
 
-DragonSlayer::DragonSlayer( std:: string name_, int hitPoints, int armor ) : Character(hitPoints, armor, 4 ), name(name_) 
+DragonSlayer::DragonSlayer( std:: string name_, int hitPoints, int armor ) : Character(getHP(), armor, 4 ), name(name_)
```
Comment (matkatmusic) [inline DragonSlayer.cpp:6]: - [ ] revise, see above comment.

File: Dwarf.cpp [hunk 13]
```diff
@@ -2,7 +2,7 @@
 #include "Character.h"
 #include "Utility.h"
 
-Dwarf::Dwarf( std::string name_, int hitpoints, int armor ) : Character(hitPoints, armor, 4), name(name_) 
+Dwarf::Dwarf( std::string name_, int hitpoints, int armor ) : Character(getHP(), armor, 4), name(name_)
```
Comment (matkatmusic) [inline Dwarf.cpp:5]: - [ ] revise, see above comment.

File: Paladin.cpp [hunk 14]
```diff
@@ -1,7 +1,7 @@
 #include "Paladin.h"
 #include "Utility.h"
 
-Paladin::Paladin( std::string name_, int hitpoints, int armor ) : Character(hitPoints, armor, 10), name(name_) 
+Paladin::Paladin( std::string name_, int hitpoints, int armor ) : Character(getHP(), armor, 10), name(name_)
```
Comment (matkatmusic) [inline Paladin.cpp:4]: - [ ] revise, see above comment.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**